orttraining/orttraining/python/training/ortmodule/__init__.py

```diff
@@ -43,6 +43,10 @@
 except ImportError as e:
     raise RuntimeError(f'PyTorch {MINIMUM_RUNTIME_PYTORCH_VERSION_STR} must be installed in order to run ONNX Runtime ORTModule frontend!') from e
 
+# PyTorch custom Autograd function support
+from ._custom_autograd_function import enable_custom_autograd_support
+enable_custom_autograd_support()
+
```

Comment:
Shouldn't this be guarded behind an enabled/disable flag as it is currently being done at orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py line 129?

Reply:
The original intention is, that flag is not used to control the custom pythonop exporter registration and autugrad function runner registration(e.g  , enable_custom_autograd_support()). Instead that is a user flag to control whether they want to fallback those PythonOp to pytorch. As the feature goes to be more stabilized I feel that flag could be removed, then the whole fallback is transparency to users. 

I am fine we put the enable_custom_autograd_support() under the flag, but as explained in the PR desc, the problem here is that flag is always false when that logic is executing. Setting that flag to be true later won't make that piece of code re-run, so enable_custom_autograd_support() never get called.

Make sense?

Reply:
Is PythonOp useful when custom autograd is disabled? If not, the lesser code ORTModule run, the lesser unexpected behavior can happen (and the faster initialization is)

When custom autograd is enabled by default, it probably makes sense to delete test_ortmodule_fallback_onnx_model__custom_autograd test, as the export for models using autograd will no longer fail, as this test asserts

Reply:
Currently Python Op is not useful when custom autograd is disabled. I understand your point, while I believe we will make the custom autograd enabled by default soon. @wschin @SherlockNoMad for more comments.

Reply:
The only concerns we enable custom function fallback is here:

We need export ops having require_grad, so set_grad_enabled to be True. If this is not a problem I can just remove flag _enable_custom_autograd_function and enable the fallback by default easily.

try:
            with torch.set_grad_enabled(self._enable_custom_autograd_function), \
                    _logger.suppress_os_stream_output(log_level=self._debug_options.logging.log_level):
                torch.onnx.export(self._flattened_module,
                                  sample_inputs_as_tuple,
                                  f,
                                  input_names=self._input_info.names,
                                  output_names=output_names,
                                  opset_version=ONNX_OPSET_VERSION,
                                  do_constant_folding=False,
                                  training=self._export_mode,
                                  dynamic_axes=self._input_info.dynamic_axes,
                                  verbose=self._debug_options.logging.log_level < LogLevel.WARNING,
                                  export_params=False,
                                  keep_initializers_as_inputs=True)
        except Exception as e:
            raise wrap_exception(ORTModuleONNXModelException,
                                 RuntimeError(f'There was an error while exporting the PyTorch model to ONNX: {e}'))
        exported_model = onnx.load_model_from_string(f.getvalue())

Reply:
I prefer this feature to be enabled by default. Most famous models define their own `autograd.Function`....